storages/request_dispatcher.go

```diff
@@ -14,60 +21,116 @@ type dispatcher interface {
 type RequestDispatcher struct {
 	Backends                  []*backend.Backend
 	syncLog                   *SyncSender
-	pickClientFactory         func(*http.Request) func([]*backend.Backend) client
-	pickResponsePickerFactory func(*http.Request) func(<-chan BackendResponse) responsePicker
+	pickClientFactory         func(*http.Request) func([]*backend.Backend, watchdog.ConsistencyWatchdog) client
+	pickResponsePickerFactory func(*Request) func(<-chan BackendResponse) responsePicker
+	watchdog                  watchdog.ConsistencyWatchdog
+	watchdogRecordFactory     watchdog.ConsistencyRecordFactory
 }
 
 // NewRequestDispatcher creates RequestDispatcher instance
-func NewRequestDispatcher(backends []*backend.Backend, syncLog *SyncSender) *RequestDispatcher {
-
+func NewRequestDispatcher(backends []*backend.Backend, syncLog *SyncSender,
+	watchdog watchdog.ConsistencyWatchdog, watchdogRecordFactory watchdog.ConsistencyRecordFactory) *RequestDispatcher {
 	return &RequestDispatcher{
 		Backends:                  backends,
 		syncLog:                   syncLog,
 		pickResponsePickerFactory: defaultResponsePickerFactory,
 		pickClientFactory:         defaultReplicationClientFactory,
+		watchdog:                  watchdog,
+		watchdogRecordFactory:     watchdogRecordFactory,
 	}
 }
 
 // Dispatch creates and calls replicators and response pickers
 func (rd *RequestDispatcher) Dispatch(request *http.Request) (*http.Response, error) {
+	storageRequest := &Request{request, nil, nil}
```

Comment:
Don't pass `zero values` in initialisation they are by deafult. BTW name each field 
&Request{request, nil, nil} -> &Request{Request: request}